Feat/TextFiled 공통컴포넌트 구현

ggoldJeongg opened the pull request:
## 🔘Part

- [x] FE


## 🔎 작업 내용

- 텍스트필드(인풋창) 공통컴포넌트 기능 구현
- size, state에 따른 (default/current/error) 스타일 변경이 가능합니다.

## 이미지 첨부
![인풋](https://github.com/user-attachments/assets/826fce36-ebe1-4f70-ae2e-376a99ba1391)

<br/>

## 🔧 앞으로의 과제
- 프로필 조회페이지 기능 구현
- 프로필 이미지 렌더링 기능
- about me 정보 조회 기능
- 카테고리 항목별 공개여부 토글 기능
- QR코드 생성 버튼 기능
- 수정 페이지 라우팅 기능
  <br/>

## ➕ 이슈 링크

- [ 소톡소톡 #10 ]

<br/>

ggoldJeongg committed:
Feat/#10 TextFiled 공통컴포넌트 구현

- 텍스트필드(인풋창창) 공통컴포넌트 기능 구현
- size, state에 따른 (default/current/error) 스타일 변경이 가능합니다.

ggoldJeongg committed:
Merge branch 'dev' into Design/10-textfiled

JunSeGue commented on src/app/components/common/TextField/styles.tsx:
```diff
@@ -0,0 +1,31 @@
+import { css } from "styled-components";
+
+export const sizeStyles = {
```
저희 css 기준 단위 px단위가 아닌 것으로 알고 잇는데 확인한번 해주세요

JunSeGue commented on src/app/components/common/TextField/TextFiled.Styled.tsx:
```diff
@@ -0,0 +1,33 @@
+  box-sizing: border-box;
+  padding: 20px;
+  border-radius: 50px;
+  font-family: var(--font-family-pretendard);
```
여기 또한 font family 를 overrite되니 지워주세요

JunSeGue commented on src/app/components/common/TextField/TextFiled.Styled.tsx:
```diff
@@ -0,0 +1,33 @@
+  .textfield-label {
+    font-size: var(--font-size-md);
+    color: var(--black-color);
+    font-family: var(--font-family-pretendard);
```
fontFamily같은 경우에 제가 'input 'inherit로 설정해놔서 따로 설정안해놔도 됩니다.

JunSeGue commented on src/types/textfield.d.ts:
```diff
@@ -0,0 +1,15 @@
+  children?: React.ReactNode;
+  name: string;
+  value?: string | number;
+  type?: "text" | "password" | "number" | "price";
```
type 같은 경우에는 email 도 추가 되어야합니다. , price같은 경우에는 저희가 안쓰는걸로 알고 있습니다.

ggoldJeongg committed:
Design/#17 코드리뷰 반영

- css : 단위 기준 px -> rem 단위로 변경
- css : font-family 중복 코드 삭제
- type : email 수정 및 추가

JunSeGue commented on src/components/common/TextField/TextFiled.Styled.tsx:
```diff
@@ -0,0 +1,31 @@
+  ${({ size }) => sizeStyles[size]}
+  ${({ state }) => stateStyles[state]}
+
+  &:focus {
```
outline : none이 되면 입력창의 선들이 사라져서 ui적으로 문제가 될 것 같습니다.

ggoldJeongg committed:
Design/#17 코드리뷰 반영

- outline 제거
- 해당 css의 기본 속성이 none값으로 확인되며, border 속성으로 focus 처리 했기때문에 필요없는 요소로 판단되어 제거합니다.
